bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/ThingFactory.java

```diff
@@ -105,7 +105,7 @@ public static Thing createThing(ThingUID thingUID, Configuration configuration,
         for (ThingHandlerFactory thingHandlerFactory : thingHandlerFactories) {
             if (thingHandlerFactory.supportsThingType(thingTypeUID)) {
                 Thing thing = thingHandlerFactory.createThing(thingTypeUID, configuration, thingUID, bridgeUID);
-                if (properties != null) {
+                if (thing != null && properties != null) {
```

Comment:
Would it be better to use:
```java
if (thing == null) {
    continue;
}
```

If there are two factories that could create the thing, you will return "null" here and do not try to use the other factory.